frontend/src/helpers/endpoint.ts

```diff
@@ -51,15 +72,25 @@ export const api = async (config: apiConfig) => {
       },
     });
 
-    if (!res.ok) {
-      return [];
-    }
+    // Parse the response body for all status codes except 204 (no content).
+    let data = res.status != 204 ? await res.json() : {};
+
+    logger.info(`[${res.status}] ${config.method}: ${res.url}}`);
 
-    return res.json();
+    // Return an ApiResponse object with status, data, and message.
+    return {
```

Comment:
Was curious why not we just return the response directly? Then we also have access to status code, status text, data, as well as other things like the response headers and etc. if we need them.

Reply:
here it returns an ApiResponse object which includes some of the mentioned values. the intended purpose is to simplify the handling after returning to the wrapper layer. for instance the response body, some request does not return any payload, so calling response.json() will throw an exception which is not that friendly for warpper functions to handle. returning ApiResponse hides some of the complexity within the endpoint layer. so this is the idea, but we can change it to return the response object directly if this approach is not working well with other services.

Reply:
Ok got it, as I don't see any potential issue for now. I am fine with your approach, so I think there is no need to change it.